internal/cmd/ske/credentials/kubeconfig/login/login.go

```diff
@@ -0,0 +1,223 @@
+			examples.NewExample(
+				"login to a SKE cluster specified in the kubeconfig",
+				"$ kubectl get pod"),
+		),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := context.Background()
+
+			model, err := parseInput()
+			if err != nil {
+				return fmt.Errorf("login SKE kubeconfig: parseInput: %w", err)
+			}
+
+			// Configure API client
+			apiClient, err := client.ConfigureClient(cmd)
+			if err != nil {
+				return err
+			}
+
+			kubeconfig := getCachedKubeConfig(model.CacheKey)
+
+			if kubeconfig == nil {
+				return getCacheAndOutputKubeconfig(ctx, cmd, apiClient, model, false, nil)
+			}
+
+			certPem, _ := pem.Decode(kubeconfig.CertData)
+			if certPem == nil {
+				_ = cache.DeleteObject(model.CacheKey)
+				return getCacheAndOutputKubeconfig(ctx, cmd, apiClient, model, false, nil)
+			}
+
+			certificate, err := x509.ParseCertificate(certPem.Bytes)
+			if err != nil {
+				_ = cache.DeleteObject(model.CacheKey)
+				return getCacheAndOutputKubeconfig(ctx, cmd, apiClient, model, false, nil)
+			}
+
+			if time.Now().After(certificate.NotAfter.UTC()) {
+				// cert expired, request new
+				_ = cache.DeleteObject(model.CacheKey)
+				return getCacheAndOutputKubeconfig(ctx, cmd, apiClient, model, false, nil)
+			} else if time.Now().Add(time.Minute * 15).After(certificate.NotAfter.UTC()) {
+				// cert expires in 15min, refresh
+				return getCacheAndOutputKubeconfig(ctx, cmd, apiClient, model, true, kubeconfig)
+			}
+
+			if err := output(cmd, model.CacheKey, kubeconfig); err != nil {
+				return err
+			}
+			return nil
+		},
+	}
+	return cmd
+}
+
+func parseInput() (*inputModel, error) {
+	obj, _, err := exec.LoadExecCredentialFromEnv()
+	if err != nil {
+		return nil, fmt.Errorf("LoadExecCredentialFromEnv: %w", err)
+	}
+
+	if err := clientauthenticationv1.AddToScheme(scheme.Scheme); err != nil {
+		return nil, err
+	}
+
+	obj, err = scheme.Scheme.ConvertToVersion(obj, clientauthenticationv1.SchemeGroupVersion)
+	if err != nil {
+		return nil, fmt.Errorf("ConvertToVersion: %w", err)
+	}
+
+	execCredential, ok := obj.(*clientauthenticationv1.ExecCredential)
+	if !ok {
+		return nil, fmt.Errorf("Conversion to ExecCredential failed")
+	}
+	if execCredential == nil || execCredential.Spec.Cluster == nil {
+		return nil, fmt.Errorf("ExecCredential contains not all needed fields")
+	}
+	config := &SKEClusterConfig{}
+	err = json.Unmarshal(execCredential.Spec.Cluster.Config.Raw, config)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshal: %w", err)
+	}
+
+	return &inputModel{
+		ClusterName: config.ClusterName,
+		ProjectId:   config.STACKITProjectID,
+		CacheKey:    fmt.Sprintf("ske-login-%x", sha256.Sum256([]byte(execCredential.Spec.Cluster.Server))),
+	}, nil
+}
+
+func buildRequest(ctx context.Context, apiClient *ske.APIClient, model *inputModel) ske.ApiCreateKubeconfigRequest {
+	req := apiClient.CreateKubeconfig(ctx, model.ProjectId, model.ClusterName)
+	expirationSeconds := "1800" // 30 min
```

Comment:
you could extract this to a const instead of hardcoding, for better maitainability